### PR TITLE
ethos: Add OS X compatibility

### PR DESCRIPTION
**Pull request by kYc0o**
This was a bit past due but here it is. However, there are some things I didn't succeed:

 - The tap device cannot be configured without an `open()` instruction, so it's not configurable until ethos open it. Thus, the original script cannot work, since the previous instructions will fail because ethos is launched at last.
A workaround is to start ethos manually and then the script, which will configure the interface and launch uhcpd.

 - I tried the gnrc_boder_router example, but I didn't succeed to route over OS X. I think the routing table is correctly configured but for some reason I cannot ping any device over the air with the prefix given by the BR. Maybe @Yonezawa-T2 has more hints?

I can mark it WIP until having the routes working, but ethos is useful for other things than the BR.

### CODE REVIEW

**Comment by smlng** [comment]
@kYc0o thanks for working on this! I'll look into it and the routing issue, too.

**Comment by aabadie** [comment]
@kYc0o, could you fix your routing issues here ?

**Comment by kYc0o** [comment]
Not yet, I'll postpone it.

**Comment by tcschmidt** [comment]
@smlng are you taking up here?

**Comment by smlng** [comment]
yep, looking at it again

File: dist/tools/ethos/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 all: ethos
 
 ethos: ethos.c
-	$(CC) -O3 -Wall ethos.c -o ethos
+	$(CC) -O3 -g3 -Wall ethos.c -o ethos
```
Comment (smlng) [inline dist/tools/ethos/Makefile]: unrelated, and not needed in production anyways?

File: dist/tools/ethos/start_network_osx.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+ETHOS_DIR="$(cd "$(dirname "$0")" && pwd -P)"
```
Comment (smlng) [inline dist/tools/ethos/start_network_osx.sh]: I think we should integrate this into one script, ie. `start_network.sh` and hide the diffs between macOS and Linux from the user in there.

Comment (kYc0o) [reply dist/tools/ethos/start_network_osx.sh]: I tried that approach but I actually wanted to have a different script for the sake of clearness. IMHO it is better to be aware when we launch a linux or a OS X script, moreover, in bash is a bit more hard to read all the needed if's for the unification.

Comment (kYc0o) [reply dist/tools/ethos/start_network_osx.sh]: Actually the reason is stated in the description of this PR.

**Comment by kYc0o** [comment]
I've rebased this PR and solved the conflicts. Everything compiles correctly but I still can't make it work correctly. I think is something related with uhcpd.

File: dist/tools/ethos/ethos.c [hunk 3]
```diff
@@ -519,13 +547,16 @@ int main(int argc, char *argv[])
 
     char ifname[IFNAMSIZ];
     strncpy(ifname, argv[1], IFNAMSIZ);
+#ifdef __linux__
```
Comment (miri64) [inline dist/tools/ethos/ethos.c]: This `#ifdef` (as well as [this](https://github.com/RIOT-OS/RIOT/pull/6718/files#diff-33a73ae3a5a86becdcf02c036d777196R534) is unnecessary, as `flags` is ignored anyway for the non `__linux__` case.

**Comment by miri64** [comment]
On the other hand #10477 + #10480 ;-)

**Comment by miri64** [comment]
Ping @kYc0o?

**Comment by kYc0o** [comment]
Unfortunately I can't work on this anymore, as I don't have an OS X. Feel free to close it, if someone wants to take over is also good.

**Comment by miri64** [comment]
OK, I close it as archived then

**Pull request closed without merging**